### PR TITLE
Refactor to monorepo

### LINKED ISSUE #134: Move the parser into its own repository
**Describe the solution you'd like**
We are planning some interesting features for xlayers but for that, we need to move the sketch parser into its own repository. 

The parser is written in pure TS so that should be easy.

### PR DESCRIPTION
**Pull request by Shikanime**
Fixes #134

### CODE REVIEW

File: packages/sketch-parser/package.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "sketch-parser",
```
[the remark this answers is not in the document]
Comment (manekinekko) [reply packages/sketch-parser/package.json]: Yep, this should be `@xlayers/sktechapp-parser`

File: tsconfig.json [hunk 2]
```diff
@@ -15,6 +15,14 @@
     "lib": [
       "es2017",
       "dom"
-    ]
+    ],
+    "paths": {
+      "parser": [
+        "dist/sketch-parser"
```
Comment (manekinekko) [inline tsconfig.json]: this should reference the src folder of the parser.

File: tsconfig.json [hunk 3]
```diff
@@ -15,6 +15,14 @@
...
+      ],
+      "parser/*": [
+        "dist/sketch-parser/*"
```
Comment (manekinekko) [inline tsconfig.json]: this should reference the src folder of the parser.

File: packages/core/src/app/core/state/page.state.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import { State, Action, StateContext, Selector } from '@ngxs/store';
-import { CodeGenKind, XlayersExporterNavBar, XlayersNgxEditorModel } from 'src/app/editor/code-editor/editor-container/codegen/codegen.service';
+import { CodeGenKind, XlayersExporterNavBar, XlayersNgxEditorModel } from '../../editor/code-editor/editor-container/codegen/codegen.service';
```
Comment (manekinekko) [inline packages/core/src/app/core/state/page.state.ts]: why not use the TS path aliases as referenced in the tsconfig?

MOTE: This also applied to the rest of the files in this PR.

File: tsconfig.json (hunk 3 above)
Comment (Jefiozie) [reply tsconfig.json]: This is probably done by the angular cli. This is default for the libary support if I'm correct. Libary's should be build before you can use it.

File: packages/core/src/app/core/state/page.state.ts (hunk 4 above)
Comment (Shikanime) [reply packages/core/src/app/core/state/page.state.ts]: I had some problems with Jest which can't import the module with an absolute path with this file structure particularly.

But yes i agree that can be a little weird.

File: packages/core/src/app/editor/viewer/lib/sketch.service.ts [hunk 5]
```diff
@@ -2,10 +2,9 @@ import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
 import { DomSanitizer } from '@angular/platform-browser';
 import { Store } from '@ngxs/store';
-import { InformUser } from 'src/app/core/state';
-import { environment } from 'src/environments/environment';
-import { SketchStyleParserService, SupportScore } from './parsers/sketch-style-parser.service';
-import { AngularSketchModule } from './sketch.module';
+import { InformUser } from '../../../core/state';
+import { environment } from '../../../../environments/environment';
+import { SketchStyleParserService, SupportScore } from '../../../../../../sketch-parser/src/public_api';
```
[the remark this answers is not in the document]
Comment (Shikanime) [reply packages/core/src/app/editor/viewer/lib/sketch.service.ts]: You are right i will look at it.

File: tsconfig.json [hunk 6]
```diff
@@ -17,11 +17,11 @@
       "dom"
     ],
     "paths": {
-      "parser": [
-        "dist/sketch-parser"
+      "@xlayers/sktechapp-parser": [
```
Comment (Jefiozie) [inline tsconfig.json]: Think a typo here it should be `@xlayers/sketchapp-parser`

**Comment by Jefiozie** [comment]
Just came up that we probably should change the `package.json`. With this implementation there should be a `pre step` that builds the parser package. Before we can run `ng serve` or `ng build` . (But maybe you where already planni)ng to do this 😃)

**Comment by Shikanime** [comment]
BTW, we need to move the `SketchData' interface to @types/sketchapp because of the circular dependency.

**Comment by Shikanime** [comment]
@manekinekko All done ! 👌

**Comment by manekinekko** [comment]
@Shikanime can you rebase plz?

**Pull request merged**